packages/react-hooks/src/hooks/toggle.ts

```diff
@@ -0,0 +1,10 @@
+import {useState, useCallback} from 'react';
+
+export function useToggle(initialState: boolean) {
+  const [state, setState] = useState(initialState);
+  const toggle = useCallback(() => setState(state => !state), []);
+
+  // cast needed to say this returns a two item array with the items in
```

Comment:
so annoying that this is the case  :<

tuple literal when?

Reply:
I found out that [`as const`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) as added in TS 3.4 does the same thing but I'm kinda paranoid about support in older versions of TS in consumers